### DISCUSSION TITLE
How do I keep track of the custom folder?

### DISCUSSION THREAD

**Discussion by ghost** [NvChad/NvChad#871] [Q&A] [answered]
I've cloned NvChad to `/config/nvim` in [my dotfiles](https://codeberg.org/LifeSymbiont/dotfiles). I want to keep track of the subfolder `/config/nvim/lua/custom` without keeping track of the rest of the nvim folder. How do I do that?

The paths are relative to my `$DOTFILES` folder. Located at `DOTFILES=~/Sync/config/dotfiles`

I've done

    cd $DOTFILES
    git submodule add https://github.com/NvChad/NvChad config/nvim

to add this to `/.gitmodules`

    [submodule "config/nvim"]
    	path = config/nvim
    	url = https://github.com/NvChad/NvChad

Then I've added this to the `/.gitignore`

    config/nvim/*
    !config/nvim/lua/
    
    config/nvim/lua/*
    !config/nvim/lua/custom/

But the custom folder is not being tracked.

I've solved it by moving the custom folder outside the submodule and linking it to it's proper place. But I'd prefer to know how to make git track it without doing that. Because this way messes the structure of my dotfiles with a folder out of place.

**Comment by siduck** [comment]
i just save the custom dir my dotfiles repo, one can just symlink it :thinking:

**Comment by ghost** [comment]
Yes, that's what I had done. I had symlinked it but I don't like that solution because it messes the structure of my dotfiles.

I install my dotfiles with symlinks. Every folder in the `$DOTFILES/config/` folder gets symlinked to `$XDG_CONFIG_HOME`. Putting the custom folder in `$DOTFILES/config/custom` would create a link for it and I don't want to create one. Putting it somewhere else doesn't make any sense because I want all configurations to be in the `$DOTFILES/config/` folder.

I will show the tree to paint a clearer picture.

```
~/Sync/config/dotfiles
❯ tree
.
├── config
│   ├── alacritty
│   ├── git
│   ├── mutt
│   ├── nvim
│   │   ├── examples
│   │   ├── init.lua
│   │   ├── LICENSE
│   │   └── lua
│   │       ├── colors
│   │       ├── core
│   │       ├── custom
│   │       └── plugins
│   ├── tig
│   ├── tmux
│   ├── zathura
│   └── zsh
└── install
    ├── arch
    ├── Brewfile
    ├── debian
    ├── helpers.sh
    ├── install.sh
    └── settings
        └── cinnamon.sh

399 directories, 1836 files
```

**Comment by ghost** [comment, marked as answer]
What I've done now is remove the `.git/` folder from NvChad and removed the submodule from my `.gitmodules`

I've added this file to a cronjob:
```
git clone https://github.com/NvChad/NvChad $DOT/config/nvim
rm -rf $DOT/config/nvim/.git $DOT/config/nvim/.gitignore $DOT/config/nvim/.github $DOT/config/nvim/examples $DOT/config/nvim/LICENSE
```

Now git is tracking the NvChad custom folder in my dotfiles without tracking the whole folder.

**Reply by siduck** [reply]
cool!
